### PR TITLE
reports: fix illegal chrs in pdf reports

### PR DESCRIPTION
**Pull request by psiinon**
## Overview
Now fixes the pdf issues, hopefully!

## Related Issues
Fixes https://github.com/zaproxy/zaproxy/issues/8330

## Checklist
- [ ] Update help
- [x] Update changelog
- [x] Run `./gradlew spotlessApply` for code formatting
- [x] Write tests
- [ ] Check code coverage
- [x] Sign-off commits
- [ ] Squash commits
- [x] Use a descriptive title

For more details, please refer to the [developer rules and guidelines](https://www.zaproxy.org/docs/developer/dev-rules-and-guidelines/).

### CODE REVIEW

File: addOns/reports/src/main/zapHomeFiles/reports/traditional-pdf/report.html [hunk 1]
```diff
@@ -179,7 +179,8 @@ <h3 th:text="#{report.alerts.list}">Alerts</h3>
 			</tr>
 			<tr th:each="alert: ${alertTree.children}">
 				<td><a th:href="'#' + ${alert.userObject.pluginId}"
-					th:text="${alert.nodeName}" href="#oluginId">Alert Name</a></td>
+					th:text="${helper.escapeXml(alert.nodeName)}" href="#oluginId">Alert
```
Comment (kingthorin) [inline addOns/reports/src/main/zapHomeFiles/reports/traditional-pdf/report.html:182]: Should the href really be olugin not plugin? I know that was there before but it seems out of place.

File: addOns/reports/src/test/java/org/zaproxy/addon/reports/ExtensionReportsUnitTest.java [hunk 2]
```diff
@@ -88,6 +88,10 @@
 
 class ExtensionReportsUnitTest extends TestUtils {
 
+    private static final String ILLEGAL_XML_CHRS = "\u0000\u0013";
+    // FIXME change everything that uses this to use ILLEGAL_XML_CHRS - these places currently fail
+    private static final String NOT_ILLEGAL_XML_CHRS = "";
```
Comment (thc202) [inline addOns/reports/src/test/java/org/zaproxy/addon/reports/ExtensionReportsUnitTest.java:93]: Can be removed now? Not used.

File: addOns/reports/src/test/java/org/zaproxy/addon/reports/ExtensionReportsUnitTest.java [hunk 3]
```diff
@@ -1220,9 +1243,8 @@ private ReportData setupReportData() {
         reportData.setSites(List.of(site1));
```
Comment (thc202) [inline addOns/reports/src/test/java/org/zaproxy/addon/reports/ExtensionReportsUnitTest.java:1243]: `site1` had `ILLEGAL_XML_CHRS`, no longer needed?

File: addOns/reports/CHANGELOG.md [hunk 4]
```diff
@@ -4,7 +4,9 @@ All notable changes to this add-on will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
 
 ## Unreleased
-
+### Fixed
```
Comment (thc202) [inline addOns/reports/CHANGELOG.md:7]: I'd mention the addressed deprecation.

File: addOns/reports/src/main/java/org/zaproxy/addon/reports/ReportDialog.java [hunk 5]
```diff
@@ -510,7 +510,9 @@ public void save() {
                                     "reports.dialog.error.generate", e.getMessage()));
             LOGGER.error(
                     "Failed to generate a report using template {}",
-                    extension.getTemplateByDisplayName(getStringValue(FIELD_TEMPLATE)),
+                    extension
+                            .getTemplateByDisplayName(getStringValue(FIELD_TEMPLATE))
+                            .getDisplayName(),
```
Comment (thc202) [inline addOns/reports/src/main/java/org/zaproxy/addon/reports/ReportDialog.java:515]: Use the name directly?

**Comment by psiinon** [comment]
Hopefully fixed all of those...

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you!

**Review by kingthorin — APPROVED** [review]
Yup :grinning:

**Comment by kingthorin** [comment]
WIP is stuck for some reason

**Comment by thc202** [comment]
Yes, having problems today.

**Pull request merged**